arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -644,7 +660,7 @@ bool RocksDBCollection::dropIndex(IndexId iid) {
           LogicalDataSource::Serialization::PersistenceWithInProgress);
 
   // log this event in the WAL and in the collection meta-data
-  res = engine.writeCreateCollectionMarker(  // write marker
+  Result res = engine.writeCreateCollectionMarker(  // write marker
```

Comment:
It looks like here we will may get one dropped index  "resurrected" with the following sequence of events:
(2 threads are dropping unrelated indexes A and B)
1.  Both threads reaches line  598 (all lines are given in NEW version)
2. Thread 1 gets writelock and removes index A from _indexes. Thread 2 stays blocked
3.  Thread 1 reaches line 657 and gets slice without index A but with index B
4. Thread 1 is blocked and thread 2 starts.
5. Thread 2 removes index B from _indexes
6. Thread 2 also reaches line 657 (with old version Thread 2 would still wait on writelock until Thread 1 completes its write)
7. Thread 2 gets slice without index A and B and goes to line 663 and writes it
8. Thread 1 is unblocked and writes slice with index B  - index B is a zombie now

Reply:
@Dronplane : true. I will try a different approach now.